### ISSUE TITLE
Support multi-class & continuous outcomes

### ISSUE THREAD

**Issue by kelly-sovacool** [SchlossLab/mikropml#154]
This is probably a bit lower on the priority list.

If more than two outcome classes, use caret multiClass and check that logit models isn't being used.

Edit: can we support continuous outcomes also?

- [x] Add multinomial logistic regression (for multiclass outcomes).
- [x] Add linear regression (for continuous outcomes).
- [x] Check to see if decision tree, random forest, and xgboost code works for multiclass and continuous; modify if needed (including check defaults to see if they're good ones to use).
- [x] Have default performance metric for each class (AUROC for binary, RMSE for continuous, logLoss for multiclass?).
~- [ ] Be clear in the vignette what models can be used for what types of outcome.~ (because all of them can be used for everything now; except maybe SVM...)
- [x] Change checks to work for non-binary data.
    - [x] Have warning for numeric data that looks like it might be binary or multiclass (related to #166).
- [x] Use `perf_metric_fn` to calculate test performance metrics.

**Comment by BTopcuoglu** [comment]
@zenalapp this might help: http://moderntoolmaking.blogspot.com/2012/07/error-metrics-for-multi-class-problems.html
https://github.com/rseiter/PracticalMLProject/blob/master/multiClassSummary.R
